media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp

```diff
@@ -3493,29 +3493,23 @@ void CodechalVdencVp9StateG12::fill_pad_with_value(PMOS_SURFACE psSurface, uint3
             return;
         }
 
-        uint8_t *src_data_y = src_data + YPlaneOffset;
+        uint32_t pad_rows = aligned_height - real_height;
 
-        uint32_t plane_size      = pitch * real_height;
-        uint32_t plane_size_full = UVPlaneOffset;
-
-        uint8_t *src_data_y_end = src_data_y + plane_size;
-
-        uint32_t y_pad_rows = aligned_height - real_height;
-        uint32_t y_pad_length = y_pad_rows * pitch;
-
-        if (src_data_y_end > src_data_y_end - y_pad_length)
-        {
-            memcpy_s(src_data_y_end, y_pad_length, src_data_y_end - y_pad_length, y_pad_length);
+        uint8_t *src_data_y     = src_data + YPlaneOffset;
+        uint32_t y_plane_size   = pitch * real_height;
+        uint8_t *src_data_y_end = src_data_y + y_plane_size;
+        uint32_t y_pitch = pitch;
+        for (int i = 0; i < pad_rows; i++) {
+          memcpy_s(src_data_y_end + i * y_pitch, y_pitch, src_data_y_end - y_pitch, y_pitch);
         }
 
-        uint32_t uv_plane_size      = (pitch * real_height)/2;
-        uint32_t uv_plane_size_full = plane_size_full / 2;
-        uint8_t *src_data_uv_end = src_data_y + plane_size_full + uv_plane_size;
-
-        if (src_data_uv_end - y_pad_length > src_data_y_end)
-        {
-            memcpy_s(src_data_uv_end, y_pad_length, src_data_uv_end - y_pad_length, y_pad_length);
-        }    
+        uint8_t *src_data_uv     = src_data + UVPlaneOffset;
+        uint32_t uv_plane_size   = pitch * real_height / 2;
+        uint8_t *src_data_uv_end = src_data_uv + uv_plane_size;
+        uint32_t uv_pitch = pitch / 2;
+        for (int i = 0; i < pad_rows; i++) {
```

Comment:
Why you do not use uint32_t for i here?
Signed int i causes warning in comparition and error if warnings are treated as errors.

Reply:
Good suggestion.